### ISSUE TITLE
Why babel load translations every request? 

### ISSUE THREAD

**Issue by WokoLiu** [python-babel/flask-babel#154]
When I use `flask_babel.gettext` function, I found that it would walk all the `BABEL_TRANSLATION_DIRECTORIES` to find and load locate&translations every request.

Is it necessary? Why we do this? Maybe we can load all the `.mo` file before every real request?

Or do I use it wrong?

Here is my `localeselector`.

```
from flask import request

@babel.localeselector
    def get_locale():
        return request.accept_languages.best
```
Please tell me I'm wrong...
Thanks

**Comment by IanRogers-Babylon** [comment]
I've been tracking through the code and came to the same conclusion (I doubted myself as it's seems pretty odd). It's this function:

> def _get_current_context():
>     if has_request_context():
>         return request
> 
>     if current_app:
>         return current_app

which could be optimised - or maybe make a subclass to override it?

**Comment by jace** [comment]
The [forked Flask-BabelEx project](https://github.com/mrjoes/flask-babelex) cites this as the reason for it to exist. Unfortunately it doesn't seem as well maintained.

**Comment by TkTech** [comment]
Resolved in v2.0.0, pending release.

**Comment by mcejp** [comment]
This is great in production, but is there a way to get the previous behavior in debug mode, for example?
